glue/clients/layer_artist.py

```diff
@@ -687,16 +691,29 @@ def __init__(self, layer, ax):
 
     def _recalc(self):
         self.clear()
+
         assert len(self.artists) == 0
 
         try:
             x = self.layer[self.xatt].ravel()
             y = self.layer[self.yatt].ravel()
+            g = self.layer[self.gatt].ravel()
         except IncompatibleAttribute as exc:
             self.disable_invalid_attributes(*exc.args)
             return False
+        self.artists = self._axes.plot(x, y, 'k.')
+
+        gu = np.unique(g)
+        colors = get_colors(len(gu))
+
+        if len(gu) is not len(g):
+            for elem, c in zip(gu, colors):
+                xg = x[g == elem]
```

Comment:
this is the kind of operation that pandas.groupby is probably much more efficient at then repeatedly testing for `g == elem`.